proposals/ranges.md

```diff
@@ -4,12 +4,30 @@
 
 This feature is about delivering two new operators that allow constructing `System.Index` and `System.Range` objects, and using them to index/slice collections at runtime.
 
-## Detailed Design
+## Overview
 
-#### System.Index
+### System.Index
 
 C# has no way of indexing a collection from the end, but rather most indexers use the "from start" notion, or do a "length - i" expression. We introduce a new Index expression that means "from the end". The feature will introduce a new unary prefix "hat" operator. Its single operand must be convertible to `System.Int32`. It will be lowered into the appropriate `System.Index` factory method call.
 
+We augment the grammar for *unary_expression* with the following additional syntax form:
+
+``` antlr
+unary_expression
```

Comment:
We need to resolve with LDM whether or not this is actually unary expression as our recollections of the matter differ ta the moment. We can clear that up when Mads is back in town.